kyaml/yaml/alias.go

```diff
@@ -43,13 +43,33 @@ var Unmarshal = yaml.Unmarshal
 var NewDecoder = yaml.NewDecoder
 var NewEncoder = func(w io.Writer) *yaml.Encoder {
 	e := yaml.NewEncoder(w)
-	e.SetIndent(indent)
-	if sequenceIndentationStyle == CompactSequenceStyle {
-		e.CompactSeqIndent()
-	}
+	e.SetIndent(DefaultMapIndent)
+	e.CompactSeqIndent()
 	return e
 }
 
+// MarshalWithIndent marshals the input interface with provided indents
+func MarshalWithIndent(in interface{}, mapIndent int, seqIndent SeqIndentType) ([]byte, error) {
```

Comment:
This reminds me if we should use https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html to pass styleOptions ? for both the new methods.

Reply:
Added EncoderOptions struct.